cmd/client/main/main.go

```diff
@@ -101,7 +101,33 @@ func main() {
 	}
 
 	if *fl.updateInfo {
-		// TODO: Send a update info message, receive the response and update the wallet balance/nonce
+		fmt.Println("Checking for new wallet info...")
+		accountInfo, err := client.RequestWalletInfo(*fl.producer)
+		if err != nil {
+			lgr.Fatalf("failed to request wallet info: %s", err.Error())
+		}
+
+		currBal, err := client.GetBalance()
+		if err != nil {
+			lgr.Fatalf("failed to get local wallet balance: %s", err.Error())
+		}
+		currNonce, err := client.GetStateNonce()
+		if err != nil {
+			lgr.Fatalf("failed to get local wallet state nonce: %s", err.Error())
+		}
+
+		if currBal != accountInfo.Balance || currNonce != accountInfo.StateNonce {
+			fmt.Println("Updating wallet info...")
+			if err := client.UpdateWallet(accountInfo.Balance, accountInfo.StateNonce); err != nil {
+				lgr.Fatalf("failed to update wallet info: %s", err.Error())
+			}
+		} else {
+			fmt.Println("Wallet is already up to date!")
+		}
+
```

Comment:
Is it necessary to compare the fields? Shouldn't `UpdateWallet` just overwrite whatever is in the wallet? I figure in the case where the fields are the same, overwriting them shouldn't be harmful so long as the private key is the same.

Reply:
I thought it is always better to avoid unnecessary overwrites, but I guess overwriting some integers here shouldn't be harmful at all. I will remove the comparisons.

Reply:
You have a point. The way I see it is, the producer will always be "correct" in this scenario, so whatever it sends is safe to use for overwriting.

Another way to look at it is we have 2 accessor function calls (`Get...`) that each open and close the wallet, and also 2 comparisons that have the potential to call `UpdateWallet` anyway. Instead we could just call `UpdateWallet`.